### PR TITLE
Issue #2962: Fix a few uninitialized variables

### PR DESCRIPTION
**Pull request by hs-apotell**
Issue #2962: Fix a few uninitialized variables

Fixing a few uninitialized variable issues identified by static analyzer
on Windows. Also, includes a few places with bad std::string_view usage.

### CODE REVIEW

File: src/Utils/StringUtils.cpp [hunk 1]
```diff
@@ -188,7 +188,9 @@ void StringUtils::replaceInTokenVector(std::vector<std::string>& tokens,
       const bool surrounded_by_quotes =
           (i > 0 && (tokens[i - 1] == "\"")) &&
           ((i < tokensSize - 1) && (tokens[i + 1] == "\""));
-      tokens[i] = surrounded_by_quotes ? removeCR(news) : news;
+      std::string tmp(news);
```
Comment (hzeller) [inline src/Utils/StringUtils.cpp:191]: the original line would be working if we converted the second branch to string (as removeCR() returns a string
```
tokens[i] = surrounded_by_quotes ? removeCR(news)  : std::string(news);
```

I suspect the static analyzer was just tripped up by the fact that the two expressions are of different type.

File: src/Utils/FileUtils.cpp [hunk 2]
```diff
@@ -238,15 +238,17 @@ fs::path FileUtils::getPreferredPath(const fs::path& path) {
...
-  std::string hashedpath;
-  std::size_t val = std::hash<std::string>{}(path.string());
   std::string last_dir = path.string();
-  if (!last_dir.empty()) last_dir.erase(last_dir.end() - 1);
-  auto it1 = std::find_if(last_dir.rbegin(), last_dir.rend(),
-                          [](char ch) { return (ch == '/' || ch == '\\'); });
-  if (it1 != last_dir.rend()) last_dir.erase(last_dir.begin(), it1.base());
-
-  hashedpath = last_dir + "_" + std::to_string(val) + separator;
+  std::size_t val = std::hash<std::string>{}(last_dir);
```
Comment (hzeller) [inline src/Utils/FileUtils.cpp:242]: Would probably a good opportunity to add a unit test for hashPath() in `FileUtils_test.cpp`

**Review by hzeller — COMMENTED** [review]
LGTM
(but see comments)

File: src/Utils/StringUtils.cpp (hunk 1 above)
Comment (hs-apotell) [reply src/Utils/StringUtils.cpp:191]: Just two different ways of keeping the compiler and the analyzer happy.

File: src/Utils/FileUtils.cpp (hunk 2 above)
Comment (hs-apotell) [reply src/Utils/FileUtils.cpp:242]: Hmm I have got my hands (and my head) full at the moment. Don't wish to sign up for anything new until I make some room on my plate. I am still investigating the regression script stalling and I am hitting walls on every direction I turn. May be next time. Sorry!

File: src/Utils/StringUtils.cpp (hunk 1 above)
Comment (hzeller) [reply src/Utils/StringUtils.cpp:191]: Yeah, but  I think `?:` is the most readable way to express this. Up to you.

Comment (hs-apotell) [reply src/Utils/StringUtils.cpp]: Done!

File: src/Utils/StringUtils.cpp [hunk 3]
```diff
@@ -182,13 +182,14 @@ void StringUtils::replaceInTokenVector(
 void StringUtils::replaceInTokenVector(std::vector<std::string>& tokens,
                                        std::string_view pattern,
                                        std::string_view news) {
+  const std::string news_s(news);
```
[the remark this answers is not in the document]
Comment (hs-apotell) [reply src/Utils/StringUtils.cpp:185]: It still complained when I initialize it with the ternary operator.

**Review by hzeller — APPROVED** [review]

**Pull request merged**